File: lib/categories/UIApplication-Additions.m
//
//  UIApplication-Additions.m
//
//
//  Created by Neil Edwards on 07/10/2010.
//  Copyright 2010 Buffer. All rights reserved.
//

#import "UIApplication-Additions.h"


static NSInteger __activityCount = 0;

@implementation UIApplication (Additions)
- (void)showNetworkActivityIndicator {
    if ( __activityCount == 0 ) {
        [self setNetworkActivityIndicatorVisible:YES];
    }
    __activityCount++;
}
- (void)hideNetworkActivityIndicator {
    __activityCount--;
    if ( __activityCount == 0 ) {
        [self setNetworkActivityIndicatorVisible:NO];
    }    
}
@end

File: CycleStreets4.0/src/extensions/UIApplicationAdditions.swift
//
//  UIApplicationAdditions.swift
//  Halo
//
//  Created by Neil Edwards on 13/02/2018.
//  Copyright © 2018 AtomicMedia. All rights reserved.
//

import Foundation
import UIKit

extension UIApplication {
	class func topViewController(_ base: UIViewController? = UIApplication.shared.keyWindow?.rootViewController) -> UIViewController? {
		if let nav = base as? UINavigationController {
			return topViewController(nav.visibleViewController)
		}
		if let tab = base as? UITabBarController {
			if let selected = tab.selectedViewController {
				return topViewController(selected)
			}
		}
		if let presented = base?.presentedViewController {
			return topViewController(presented)
		}
		return base
	}
}
